service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrFolderEffectiveTimeTest.java

```diff
@@ -0,0 +1,176 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.util.stream.Stream;
+
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Period;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+
+public class EhrFolderEffectiveTimeTest {
+
+    private static final FhirParseService FHIR_PARSER = new FhirParseService();
+
+    private static final String INITIAL_START_DATE_HL7 = "20190828103005";
+    private static final String EARLIER_START_DATE_HL7 = "20180828103005";
+    private static final String INITIAL_END_DATE_HL7 = "20200828103005";
+    private static final String LATER_END_DATE_HL7 = "20210828103005";
+
+    private static final String INITIAL_START = "2019-08-28T10:30:05+01:00";
+    private static final String EARLIER_START = "2018-08-28T10:30:05+01:00";
+    private static final String LATER_START = "2020-08-28T10:30:05+01:00";
+    private static final String INITIAL_END = "2020-08-28T10:30:05+01:00";
+    private static final String EARLIER_END = "2019-08-28T10:30:05+01:00";
+    private static final String LATER_END = "2021-08-28T10:30:05+01:00";
+
+    private static final String ENCOUNTER_WITH_FULL_DATES = "{\"resourceType\": \"Encounter\", "
+        + "\"period\": { \"start\": \"%s\", \"end\": \"%s\"}}";
+    private static final String ENCOUNTER_WITH_START_DATE_ONLY = "{\"resourceType\": \"Encounter\", \"period\": {\"start\": \"%s\"}}";
+    private static final String ENCOUNTER_WITH_END_DATE_ONLY = "{\"resourceType\": \"Encounter\", \"period\": {\"end\": \"%s\"}}";
+
+    @Test
+    public void When_UpdatingEffectiveTimePeriodWithStartAndEndDateForFirstTime_Expect_BothSet() {
+        Period initialPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, INITIAL_START, INITIAL_END));
+
+        EhrFolderEffectiveTime ehrFolderEffectiveTime = new EhrFolderEffectiveTime();
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(initialPeriod);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().get()).isEqualTo(INITIAL_START_DATE_HL7);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().get()).isEqualTo(INITIAL_END_DATE_HL7);
```

Comment:
A few things can be done here, to make use of both jupiter and assertj vocabulary

First, we'd like to know everything that's wrong instead of just the first failed assertion, so I suggest using jupiter's `assertAll()`.

Then, assertj understands optionals so you don't need to evaluate them and then compare the results. `assertThat(optional).isPresent()`, and `assertThat(optional).contains(expected)`.

And then we can note that `.contains` also covers the presence test anyway:
```suggestion
        assertAll(
            () -> assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow()).contains(INITIAL_START_DATE_HL7),
            () -> assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh()).contains(INITIAL_END_DATE_HL7)
        );
```